source/customization/menus.rst

```diff
@@ -1,5 +1,367 @@
+               { "command": "undo", "mnemonic": "U" },
+               { "command": "redo_or_repeat", "mnemonic": "R" },
+               {
+                   "caption": "Undo Selection",
+                   "children":
+                   [
+                       { "command": "soft_undo" },
+                       { "command": "soft_redo" }
+                   ]
+               },
+               { "caption": "-", "id": "clipboard" },
+               { "command": "copy", "mnemonic": "C" },
+               { "command": "cut", "mnemonic": "t" },
+               { "command": "paste", "mnemonic": "P" },
+               { "command": "paste_and_indent", "mnemonic": "I" },
+               { "command": "paste_from_history", "caption": "Paste from History" }
+           ]
+       }
+   ]
+
+
+Images
+******
+
+.. figure:: images/context_menu_default.png
+   :class: shadowed
+   :figclass: float-left
+   :figwidth: 50%
+
+   The default context menu in the editing area.
+
+.. figure:: images/context_menu_modified.png
+   :class: shadowed
+   :figclass: float-left
+   :figwidth: 50%
+
+   A modified context menu in the editing area.
+
+
+.. _menu-types:
+
+Available Menus
+===============
+
+The file name
+of a ``.sublime-menu`` file
+specifies the menu in the application
+to be modified.
+
+The following menus are available:
+
+========================   ===============
+     File/Menu Name          Description
+========================   ===============
+**Main**                   Main menu
+
+**Context**                Context menu in the editing area
+
+**Find in Files**          Appears when clicking the "…" button
+                           in the :doc:`Find in Files
+                           </search_and_replace/search_and_replace_files>` panel
+
+**Side Bar**               Context menu for each node in the sidebar
+
+**Side Bar Mount Point**   Additional context menu items
+                           for the top-level nodes in the sidebar
+
+**Tab Context**            Context menu of the tab bar
+
+**Widget Context**         Context menu of input fields
+                           in all kinds of widgets,
+                           including Command Palette, Goto Anything,
+                           the find panels
+                           and those opened by APIs
+========================   ===============
+
+Additionally,
+the following four menus open
+when their column in the status bar
+is clicked:
+
+- **Encoding**
+- **Line Endings**
+- **Indentation**
+- **Syntax**
+
+.. figure:: images/statusbar_menu.gif
+   :class: shadowed
+
+   Demonstration of a status bar menu.
+
+
+.. _menu-items:
+
+Menu Items
+==========
+
+A menu item can either
+invoke a command (with arguments)
+when it is selected
+or have a sub-menu.
+
+The available properties are:
+
+- a command name,
+- arguments for the command,
+- an ID,
+- a caption,
+- a mnemonic and
+- a sub-menu.
+
+In order to function properly,
+it must provide *at least* either out of

```

Comment:
Or maybe it does, but seems unusual to me and makes me trip on it. Maybe it's just me.